test-pup/example-react-app/form-controls/checkbox-field.test.js

```diff
@@ -0,0 +1,37 @@
+const puppeteer = require("puppeteer");
+const {login} = require("../../common/login-to-scr");
+const {getDocument, queries} = require("pptr-testing-library");
+const {getSubmitResult} = require("../../common/custom-form-controls");
+const {findByLabelText} = queries;
+
+describe('CheckboxField form control', () => {
+    let page;
+    const url = 'customFormControls';
+
+    beforeAll(async () => {
+        const browser = await puppeteer.launch();
+        page = await browser.newPage();
+        await login(page);
+    });
+
+    afterAll(async done => {
+        page.browser().close();
+        done();
+    });
+
+    it('Checkbox should return to initial state after two clicks', async () => {
+        await page.goto(`http://localhost:3000/${url}`);
+        const $document = await getDocument(page);
+
+        const $field = await findByLabelText($document, 'Wheel On Right');
+        await $field.evaluate(checkbox => checkbox.checked = false);
```

Comment:
I think it would make sense to initialize the form with some values before checking what field components display. You should be able to do it with something like this:

```
useEffect(() => {
  form.setFieldsValue({
    manufacturer: 'Initial Manufacturer',
    wheelOnRight: true,
    // . . .
  });
});
```

Reply:
I used initialValues Form props for it. Main difference, i think, would be noticeable if we call `resetFields` - function reset fields to initial values